addons/udes_stock/tests/test_refactoring_splitting.py

```diff
@@ -918,8 +926,129 @@ def test_split_mixed_products_partial_reserve(self):
+
+        putaway_picking = self.pick_1.u_next_picking_ids
+
+        # put away is not done, so no date done should be set
+        self.assertNotEqual(putaway_picking.state, "done")
+        self.assertFalse(putaway_picking.date_done)
+
+    def test_does_not_propagate_done_date_to_back_order(self):
+        """If we complete a pick with some lines partially completed, generated ones don't have a done date"""
+
+        move_line_id_1, move_line_id_2 = self.pick_2.move_line_ids
+        move_line_id_1.write({"location_dest_id": self.received_location.id, "qty_done": 10})
+        move_line_id_2.write({"location_dest_id": self.received_location.id, "qty_done": 1})
+        self.pick_2.action_done()
+
+        backorder = self.pick_2.backorder_id
+
+        # check that pick_2 is done and has a date done, and backorder is not done and does not have a date done
+        self.assertEqual(self.pick_2.state, "done")
+        self.assertTrue(self.pick_2.date_done)
+        self.assertNotEqual(backorder.state, "done")
+        self.assertFalse(backorder.date_done)
+
+        putaway_picking = self.pick_2.u_next_picking_ids
+
+        # check that putaway is not in state done and does not have date done set
+        self.assertNotEqual(putaway_picking.state, "done")
+        self.assertFalse(putaway_picking.date_done)
+
+    def test_post_validate_refactor(self):
+        """Test123"""
+
+        mv = self.create_move(self.banana, 1, self.pick_1)
+        pickings = self.pick_1 | self.pick_2
+        self.pick_1.action_assign()
+        self.assertEqual(len(pickings.mapped("move_lines")), 4)
+        for move_line in pickings.mapped("move_line_ids"):
+            move_line.write(
+                {"location_dest_id": self.received_location.id, "qty_done": move_line.product_qty}
+            )
+        pickings.action_done()
+        self.assertEqual(self.pick_1.date_done, self.pick_2.date_done)
+
+        # Group by product post validation
+        self.picking_type_in.write(
+            {
+                "u_post_validate_action": "group_by_move_key",
+                "u_move_key_format": "{product_id.id}",
+            }
+        )
+        # Change the time shift of the apple move and picking time,
+        # so any new existing refactored pickings with apples have the greatest date,
+        # Or at least greater than bananas.
+        time_shift = datetime.datetime.now() + datetime.timedelta(hours=1)
+        self.pick_1.date_done = time_shift
+        self.pick_1.move_lines.filtered(lambda mv: mv.product_id == self.apple).write({"date": time_shift})
+
+        self.assertGreater(self.pick_1.date_done, self.pick_2.date_done)
+
+        # Call refactor
+        pickings.mapped("move_lines")._action_refactor(stage="validate")
+
+        all_pickings = self.env["stock.picking"].search([("picking_type_id", "=", self.picking_type_in.id)])
+        draft_pickings = all_pickings.filtered(lambda p: p.state == "draft")
```

Comment:
Might want to add something like 
self.assertTrue(set(pickings).issubset(draft_pickings)) 
to show that the refactoring that is being done is moving everything into nnew pickings
Or just add a comment